crates/astria-conductor/src/executor/mod.rs

```diff
@@ -113,9 +113,204 @@ impl From<SequencerBlockData> for ExecutorCommand {
+            .await
+            .wrap_err("executor failed to get commitment state")?;
+
+        info!(
+            soft_block_hash = hex::encode(&commitment_state.soft.hash),
+            firm_block_hash = hex::encode(&commitment_state.firm.hash),
+            "initial execution commitment state",
+        );
+
+        let Some(executable_block_height) = calculate_executable_block_height(
+            sequencer_height_with_first_rollup_block,
+            commitment_state.soft.number,
+        ) else {
+            bail!(
+                "encountered overflow when calculating first executable block height; sequencer \
+                 height with first rollup block: {sequencer_height_with_first_rollup_block}, \
+                 height recorded in soft commitment state: {}",
+                commitment_state.soft.number,
+            );
+        };
+
+        Ok(Executor {
+            block_channel,
+            shutdown,
+            execution_rpc_client,
+            chain_id,
+            commitment_state,
+            executable_block_height,
+            sequencer_hash_to_execution_block: HashMap::new(),
+            pre_execution_hook,
+        })
+    }
+}
+
+impl<TChainId, TBlockChannel, TRollupAddress, TShutdown>
+    ExecutorBuilder<TChainId, TBlockChannel, TRollupAddress, TShutdown>
+{
+    pub(crate) fn chain_id(
+        self,
+        chain_id: ChainId,
+    ) -> ExecutorBuilder<WithChainId, TBlockChannel, TRollupAddress, TShutdown> {
+        let Self {
+            block_channel,
+            optimism_hook,
+            sequencer_height_with_first_rollup_block,
+            rollup_address,
+            shutdown,
+            ..
+        } = self;
+        ExecutorBuilder {
+            chain_id: WithChainId(chain_id),
+            block_channel,
+            optimism_hook,
+            sequencer_height_with_first_rollup_block,
+            rollup_address,
+            shutdown,
+        }
+    }
+
+    pub(crate) fn block_channel(
+        self,
+        block_channel: Receiver,
+    ) -> ExecutorBuilder<TChainId, WithBlockChannel, TRollupAddress, TShutdown> {
+        let Self {
+            chain_id,
+            optimism_hook,
+            sequencer_height_with_first_rollup_block,
+            rollup_address,
+            shutdown,
+            ..
+        } = self;
+        ExecutorBuilder {
+            chain_id,
+            block_channel: WithBlockChannel(block_channel),
+            optimism_hook,
+            sequencer_height_with_first_rollup_block,
+            rollup_address,
+            shutdown,
+        }
+    }
+
+    pub(crate) fn sequencer_height_with_first_rollup_block(
+        mut self,
+        sequencer_height_with_first_rollup_block: u32,
+    ) -> Self {
+        self.sequencer_height_with_first_rollup_block = sequencer_height_with_first_rollup_block;
+        self
+    }
+
+    pub(crate) fn set_optimism_hook(mut self, handler: Option<optimism::Handler>) -> Self {
```

Comment:
Note that this follows the ecosystem using `set_*` for setters that replace builder pattern fields by a full type.

I would provide `fn optimism_hook(self, handler: Handler)` which would then `self.optimism_hook.replace(handler);`, but since this is not a public interface it would get flagged as never used.